packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -496,6 +498,17 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
         }
     }
 
+    /**
+     * Called when a new reference is added to another Fluid object. This is required so that garbage collection can
+     * identify all references added in the system.
+     * @param id - The id of the node that added the reference.
```

Comment:
Usage of ID seems rather limiting here. In a sense that this interface will not support well deeper tree structure (data stores on channels). Should it be a handle? Or a GC node path (almost the same, not sure if we differentiate the two)?
That way I think we will not need to have prefixAndNormalizeId() as well

Reply:
I was trying to use a similar structure to how we get the GC data - Each node returns a set of nodeId -> outboundRoutes, where nodeId can be the id of nodes inside that node or its "/" indicating that object itself.
I could call this `nodeId` and add comments explaining that this means the id of a GC node which could be the node itself or sub-nodes.

I don't like passing a handle because today data stores do not have handles. So, we will have to create a new handle every time or add an internal handle for data stores. I am not sure if that's a good idea.

Reply:
Changed this to use handles for the source node as well.